server/src/main/java/org/opensearch/gateway/GatewayAllocator.java

```diff
@@ -76,8 +76,8 @@ public class GatewayAllocator implements ExistingShardsAllocator {
 
     private final RerouteService rerouteService;
 
-    private final PrimaryShardAllocator primaryShardAllocator;
-    private final ReplicaShardAllocator replicaShardAllocator;
+    private PrimaryShardAllocator primaryShardAllocator;
+    private ReplicaShardAllocator replicaShardAllocator;
```

Comment:
As discussed, let us add back the `final` modifier and instead of a UT, we can use an Integration test for `explainUnassigned` in batch mode scenario.